File: src/ai/local.ts
export async function chatLocal(
  prompt: string,
  baseUrl: string,
  model: string,
): Promise<string> {
  const url = `${baseUrl.replace(/\/$/, "")}/api/chat`;

  const response = await fetch(url, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      model,
      messages: [
        {
          role: "system",
          content: "You are a helpful search result summarizer. Be concise and factual.",
        },
        {
          role: "user",
          content: prompt,
        },
      ],
      stream: false,
    }),
    signal: AbortSignal.timeout(30_000),
  });

  if (!response.ok) {
    throw new Error(`Local model returned HTTP ${response.status}`);
  }

  const data = (await response.json()) as {
    message?: { content?: string };
  };

  const content = data.message?.content;
  if (!content) {
    throw new Error("Local model returned empty response");
  }

  return content.trim();
}

File: src/evoscry/ai/local.py
"""Ollama-compatible local model client."""

from __future__ import annotations

import httpx


async def chat_local(prompt: str, base_url: str, model: str) -> str:
    """Call a local Ollama-compatible model."""
    url = f"{base_url.rstrip('/')}/api/chat"

    async with httpx.AsyncClient(timeout=30.0) as client:
        resp = await client.post(
            url,
            json={
                "model": model,
                "messages": [
                    {
                        "role": "system",
                        "content": "You are a helpful search result summarizer. Be concise and factual.",
                    },
                    {"role": "user", "content": prompt},
                ],
                "stream": False,
            },
        )
        resp.raise_for_status()
        data = resp.json()

    content = data.get("message", {}).get("content")
    if not content:
        raise RuntimeError("Local model returned empty response")

    return content.strip()
